src/index.ts

```diff
@@ -143,9 +143,42 @@ export class WalletPluginAnchor implements WalletPlugin {
      */
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     async sign(resolved: ResolvedSigningRequest, context: TransactContext): Promise<Signature> {
+        // Create the identity request to be presented to the user
+        const {callback, request} = await createTransactionRequest(context)
+
+        // Tell Wharf we need to prompt the user with a QR code and a button
+        context.ui.prompt({
```

Comment:
This should probably just be a `context.ui.status` message instead of a prompt. The UI we want to create is going to end up being similar to this:

<img width="738" alt="image" src="https://user-images.githubusercontent.com/677686/218224713-cdfe3329-485c-4566-9d6c-d0f582e9fe5d.png">

We may need to create a "timer" element or something... not sure yet on this.

Reply:
I've added a new prompt type in the session kit `0.3.0-ui-7` called `countdown`, and support for it in the web-ui-renderer in 0.0.4. You can now create a countdown like so:

```ts
context.ui.prompt({
    title: 'Sign',
    body: 'Please open the Anchor Wallet on "DEVICE" to review and sign the transaction.',
    elements: [
        {
            type: 'countdown',
            data: resolved.transaction.expiration.toDate().toISOString(),
        },
        {
            type: 'button',
            label: 'Sign manually or with another device',
            data: String(resolved),
        },
    ],
})
```

You'll have to create a Promise to await in the plugin itself, but the above shows how to trigger the UI to start the countdown based on the transactions expiration. 

It'll be a promise race between the above timeout against the expiration of the transaction, and the promise returned by the wallet.

Reply:
Awesome! I'll use that!